jobbergate-agent/jobbergate_agent/jobbergate/submit.py

```diff
@@ -244,7 +244,9 @@ async def _reject_handler(params: DoExceptParams) -> None:
         logger.debug(f"Using local slurm user '{username}' for job submission")
         subprocess_handler = SubprocessAsUserHandler(username)
 
-    submit_dir = pending_job_submission.execution_directory or SETTINGS.DEFAULT_SLURM_WORK_DIR
+    submit_dir = pending_job_submission.execution_directory or Path(
+        SETTINGS.DEFAULT_SLURM_WORK_DIR.format(username=username)
```

Comment:
As I said in another comment, we should remove the `DEFAULT_SLURM_WORK_DIR` entirely and use `Path.home()` here. We don't actually have a use-case that I know of for using `DEFAULT_SLURM_WORK_DIR`. Simpler is better when we can have it.

Reply:
We got to an agreement on the comment https://github.com/omnivector-solutions/jobbergate/pull/732#discussion_r1958203915.